File: python/lista1/cd.py
from memoria_s import MemoriaS

class CD(MemoriaS):
    ABERTO = 0
    FECHADO = 1

    def __init__(self, new_total, new_unidade):
        super().__init__(new_total, new_unidade)
        self.__estado = CD.ABERTO

    def get_perda(self):
        return 0.98

    def get_espaco_disponivel_real_kb(self):
        return self._get_espaco_disponivel_kb() * self.get_perda()

    def grava_kb(self, new_tamanho):
        if self.__estado == CD.ABERTO:
            gravou = super().grava_kb(new_tamanho)
            if gravou:
                self.__estado = CD.FECHADO
            return gravou
        return False

    def get_estado(self):
        if self.__estado == CD.ABERTO:
            return "ABERTO"
        return "FECHADO"

    def __str__(self):
        return f"CD Estado {self.get_estado()} " + super().__str__()

File: java/exercicios/lista1/CD.java
package lista1;

public class CD extends MemoriaS {
    public static final int ABERTO  = 0;
    public static final int FECHADO = 1;
    private int estado;

    public CD(int newTotal, int newUnidade) {
        super(newTotal, newUnidade);
        this.estado = ABERTO;
    }

    @Override
    public double getPerda() {
        return 0.98;
    }
    
    @Override
    public double getEspacoDisponivelRealKB() {
        return getEspacoDisponivelKB() * getPerda();
    }

    @Override
    public boolean GravaKB(int newTamanho) {
        if (estado == ABERTO) {
            boolean gravou = super.GravaKB(newTamanho);
            if (gravou) {
                estado = FECHADO;
            }
            return gravou;
        }
        return false;
    }

    public String getEstado() {
        if (estado == ABERTO) {
            return "ABERTO";
        }
        return "FECHADO";
    }

    @Override
    public String toString() {
        return "CD Estado " + getEstado() + " "
             + super.toString();
    }
}
